src/components/pages/Triggers/TriggersList/AddTriggersModal/AddTriggerModal.tsx

```diff
@@ -0,0 +1,179 @@
+  const [firstStepValues, setFirstStepValues] = useState<Record<string, string | Option[]>>({});
+
+  const topRef = useRef<HTMLDivElement>(null);
+  const inTopInViewport = useInViewport(topRef);
+
+  const [form] = Form.useForm();
+
+  const onFinish = () => {
+    const values = form.getFieldsValue();
+
+    const resourceSelector = getResourceIdentifierSelector(
+      firstStepValues.resourceLabelSelector || firstStepValues.resourceNameSelector,
+      appNamespace
+    );
+
+    const testSelector = getResourceIdentifierSelector(
+      values.testLabelSelector || values.testNameSelector,
+      appNamespace
+    );
+
+    const [action, execution] = values.action.split(' ');
+
+    const body = {
+      ...(name ? {name} : {}),
+      resource: firstStepValues.resource,
+      event: firstStepValues.event,
+      action,
+      execution,
+      testSelector,
+      resourceSelector,
+    };
+    createTrigger(body)
+      .then(res => displayDefaultNotificationFlow(res))
+      .then(res => {
+        if (res && 'data' in res) {
+          navigate(`/triggers/${res.data.name}`);
+        }
+      })
+      .catch(err => {
+        setError(err);
+
+        if (!inTopInViewport && topRef && topRef.current) {
+          topRef.current.scrollIntoView();
+        }
+      });
+  };
+
+  return (
+    <>
+      <div ref={topRef} />
+      <Space size={24} direction="vertical" style={{width: '100%', marginTop: 0}}>
+        <Space direction="vertical" style={{width: '100%'}}>
+          {error ? (
+            <StyledNotificationContainer>
+              <NotificationContent status="failed" message={error.message} title={error.title} />
+            </StyledNotificationContainer>
+          ) : null}
+          <Text className="regular middle">Name</Text>
+          <Input
+            placeholder="e.g.: container-deployment-xyz"
+            value={name}
+            onChange={event => {
+              setName(event.target.value);
+            }}
+          />
+        </Space>
+        <Steps current={currentStep} items={[{title: 'Condition'}, {title: 'Action'}]} />
+        <Form layout="vertical" onFinish={onFinish} form={form} name="add-trigger-form">
+          {currentStep === 0 ? (
+            <>
+              <StyledStepDescription>
+                <Text color={Colors.slate400} className="regular middle">
+                  Define the conditions to be met for the trigger to be called.
+                </Text>
+              </StyledStepDescription>
+              <ConditionFormItems />
+              <Form.Item
+                style={{
+                  textAlign: 'end',
+                  marginBottom: 0,
+                }}
+                shouldUpdate
+              >
+                {({getFieldsValue, validateFields}) => (
+                  <Button
+                    $customType="primary"
+                    onClick={() => {
+                      validateFields().then(() => {
+                        setCurrentStep(1);
```

Comment:
If `0` and `1` are constants here, it should be rather `enum` or `const`

Reply:
Ok, i will change it to enum